### DISCUSSION TITLE
Store&Forward

### DISCUSSION THREAD

**Discussion by NguyenKhanh27** [edgexfoundry/community#81] [Q&A] [answered]
Hi all, I wanna know deeper about Store and Forward ability. Regarding to the stored data, it would be stored in redis(as default) until it is sent  or exceed the retry count. Am I thinking it on a right way? Additionally, I think it is stored in core-data in form of an event(but I cant find any of these in document T.T). I would appreciate if these thinking could be confirmed or fixed into right way.
Im using version 2.3 btw

**Comment by lenny-goodell** [comment, marked as answer]
Correct, Store and Forward data is stored in Redis by default and the source Event/Reading data `MAY` also be store in Core Data. Note that Core data isn't required, thus it may not be stored. Also the Store and Forward data isn't necessarily the source Event/Reading data. It may, have been processed (transformed, compressed, encrypted, etc.) prior to failing export and be stored for later retry. 

Here is the documentation for how Store and Forward works. See the 2nd note about data transformations.
https://docs.edgexfoundry.org/2.3/microservices/application/AdvancedTopics/#how-it-works
